packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNav/useTeachingPopoverCarouselNav.tsx

```diff
@@ -21,6 +22,31 @@ export const useTeachingPopoverCarouselNav_unstable = (
   const focusableGroupAttr = useFocusableGroup({ tabBehavior: 'limited' });
   const totalPages = useTeachingPopoverCarouselContext_unstable(context => context.totalPages);
   const currentPage = useTeachingPopoverCarouselContext_unstable(context => context.currentPage);
+  const setCurrentPage = useTeachingPopoverCarouselContext_unstable(context => context.setCurrentPage);
+
+  let handleKeyDown = useEventCallback((ev: React.KeyboardEvent<HTMLDivElement>) => {
+    props.onKeyDown?.(ev);
+
+    if (ev.isDefaultPrevented()) {
+      return;
+    }
+
+    const key = ev.key;
+
+    if (key === ArrowUp || key === ArrowLeft) {
```

Comment:
Just a nitpick but you can collapse this code a bit:

```ts
let nextPage = currentPage;
if (key === ArrowUp || key === ArrowLeft) {
  nextPage = Math.max(0, currentPage - 1);
} else if (key === ArrowDown || key === ArrowRight) {
  nextPage = Math.min(totalPages - 1, currentPage + 1);
}

setCurrentPage(nextPage);
```

Just double check that the call to `setCurrentPage()` doesn't trigger a re-render when `nextPage === currentPage`.